packages/manager-react-components/src/hooks/services/services.type.ts

```diff
@@ -0,0 +1,151 @@
+export type EndRuleStrategy =
+  | 'CANCEL_SERVICE'
+  | 'REACTIVATE_ENGAGEMENT'
+  | 'STOP_ENGAGEMENT_FALLBACK_DEFAULT_PRICE'
+  | 'STOP_ENGAGEMENT_KEEP_PRICE';
+
+export type LifecycleAction =
+  | 'earlyRenewal'
+  | 'terminate'
+  | 'terminateAtEngagementDate'
+  | 'terminateAtExpirationDate';
+
+export type RenewMode = 'automatic' | 'manual';
+
+export type CustomerContact = {
+  customerCode: string;
+  type: 'administrator' | 'billing' | 'technical';
+};
+
+export type ResourceStatus =
+  | 'active'
+  | 'deleted'
+  | 'suspended'
+  | 'toActivate'
+  | 'toDelete'
+  | 'toSuspend';
+
+export type ServiceDetails = {
+  billing: {
+    engagement: {
+      endDate: string | null;
+      /**
+       * Describes the rule applied at the end of the Engagement
+       */
+      endRule: {
+        possibleStrategies: EndRuleStrategy[];
+        strategy: EndRuleStrategy;
+      };
+    };
+    engagementRequest: {
+      pricingMode: string;
+      requestDate: string;
+    };
+    expirationDate: string;
+    group: {
+      id: number;
+    };
+    lifecycle: {
+      capacities: {
+        actions: LifecycleAction[];
+      };
+      current: {
+        createDate: string;
+        pendingActions: LifecycleAction[];
+        state:
+          | 'active'
+          | 'error'
+          | 'rupture'
+          | 'terminated'
+          | 'toRenew'
+          | 'unpaid'
+          | 'unrenewed';
```

Comment:
why not extract the remaning enums as their own types like your did for `ResourceStatus` or `LifecycleAction` for example ?

Reply:
Done